psi4/driver/schema_wrapper.py

```diff
@@ -586,6 +586,18 @@ def run_json_qcschema(json_data, clean, json_serialization, keep_wfn=False):
             )):
                 json_data["extras"]["qcvars"][k] = _serial_translation(v, json=json_serialization)
 
+    # Add in handling of matrix arguments which need to be obtained by a
+    # a function call.
+
+    if json_data["extras"].get("psi4:save_tamps", False):
+        if type(wfn.reference_wavefunction()) is psi4.core.RHF :
+            json_data["extras"]["psi4:arrays"] = {}
+            json_data["extras"]["psi4:arrays"]["tIjAb"] = wfn.get_amplitudes()["tIjAb"].to_array().tolist()
+            json_data["extras"]["psi4:arrays"]["tIA"] = wfn.get_amplitudes()["tIA"].to_array().tolist()
+            if json_data["model"]["method"].lower() in ["ccsd", "scf"] :
```

Comment:
do you perhaps want the method filter right after the RHF filter? and won't the get_amplitudes calls have failed by this point for method=scf? maybe `"psi4:tamps"` instead of `"psi4:arrays"`? wfn.arrays() returns things gradient- and dipole-sized.

Reply:
Fixed